### PR TITLE
Comments for better understanding of zero stage1_2 

### PR DESCRIPTION
**Pull request by kisseternity**
Besides, comment some unnecessary variables

### CODE REVIEW

File: deepspeed/runtime/zero/stage_1_and_2.py [hunk 1]
```diff
@@ -356,7 +358,9 @@ def __init__(self,
                     self.parallel_partitioned_bit16_groups[i][partition_id].to(
                         self.device).clone().half().detach())
 
-            # modify optimizer of have flat master weight
+            # modify optimizer of have flat master weight.
```
Comment (tjruwase) [inline deepspeed/runtime/zero/stage_1_and_2.py:361]: Thanks for this documentation improvements. However, the first line here seems to have a typo. Can you please fix? Thanks!

File: deepspeed/runtime/zero/stage_1_and_2.py [hunk 2]
```diff
@@ -1426,7 +1430,7 @@ def get_data_parallel_partitions(self, tensor, group_id):
         partitions = []
 
         dp = dist.get_world_size(group=self.real_dp_process_group[group_id])
-        dp_id = dist.get_rank(group=self.real_dp_process_group[group_id])
+        # dp_id = dist.get_rank(group=self.real_dp_process_group[group_id])
```
Comment (tjruwase) [inline deepspeed/runtime/zero/stage_1_and_2.py:1433]: It is okay to remove this code if `dp_id` is not used.

File: deepspeed/runtime/zero/stage_1_and_2.py [hunk 3]
```diff
@@ -1691,7 +1695,7 @@ def step(self, closure=None):
         self.start_timers([OPTIMIZER_GRADIENTS])
         norm_groups = []
         single_partition_grad_groups = []
-        skip = False
+        # skip = False
```
Comment (tjruwase) [inline deepspeed/runtime/zero/stage_1_and_2.py:1698]: It is okay to remove `skip` if not used.

**Review by tjruwase — APPROVED** [review]

File: deepspeed/runtime/zero/stage_1_and_2.py (hunk 1 above)
Comment (kisseternity) [reply deepspeed/runtime/zero/stage_1_and_2.py]: Ok, I've re-commented this.

Comment (kisseternity) [reply deepspeed/runtime/zero/stage_1_and_2.py]: > Thanks for this documentation improvements. However, the first line here seems to have a typo. Can you please fix? Thanks!

Ok, I've re-commented this.

File: deepspeed/runtime/zero/stage_1_and_2.py (hunk 2 above)
Comment (kisseternity) [reply deepspeed/runtime/zero/stage_1_and_2.py:1433]: > It is okay to remove this code if `dp_id` is not used.

It's not used, just comment it for safe.

File: deepspeed/runtime/zero/stage_1_and_2.py (hunk 3 above)
Comment (kisseternity) [reply deepspeed/runtime/zero/stage_1_and_2.py:1698]: > It is okay to remove `skip` if not used.

It's not used, just comment it for safe.

**Pull request merged**